qiskit/circuit/parameterexpression.py

```diff
@@ -90,7 +91,7 @@ def bind(self, parameter_values: Dict) -> 'ParameterExpression':
         """
 
         self._raise_if_passed_unknown_parameters(parameter_values.keys())
-        self._raise_if_passed_non_real_value(parameter_values)
+        # self._raise_if_passed_non_real_value(parameter_values)
```

Comment:
We probably still want a relaxed version of `_raise_if_passed_non_real_value` to raise on non-numeric types.

Reply:
If we follow Ali's rec, this can remain, right?

Reply:
The error message in that check might have to be updated, too.